File: spring-boot-postgres/src/test/java/de/larmic/postgres/database/TweetRepositoryIT.java
package de.larmic.postgres.database;

import de.larmic.postgres.database.model.TweetEntity;
import de.larmic.postgres.testcontainers.AbstractIntegrationTest;
import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.within;

class TweetRepositoryIT extends AbstractIntegrationTest {

    @Test
    void save() {
        final var entity = new TweetEntity("Some test message");

        this.tweetRepository.save(entity);

        assertThat(entity.getId()).isNotZero();
        assertThat(entity.getMessage()).isEqualTo("Some test message");
        assertThat(entity.getCreateDate()).isCloseTo(LocalDateTime.now(), within(1, ChronoUnit.SECONDS));
        assertThat(entity.getLastUpdateDate()).isCloseTo(LocalDateTime.now(), within(1, ChronoUnit.SECONDS));
    }
}

File: spring-boot-postgres-kotlin/src/test/kotlin/de/larmic/postgres/database/TweetRepositoryIT.kt
package de.larmic.postgres.database

import de.larmic.postgres.database.model.TweetEntity
import de.larmic.postgres.testcontainers.AbstractIntegrationTest
import org.assertj.core.api.Assertions
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.time.LocalDateTime
import java.time.temporal.ChronoUnit

internal class TweetRepositoryIT : AbstractIntegrationTest() {

    @Test
    fun save() {
        val entity = TweetEntity(message = "Some test message")

        tweetRepository.save(entity)

        assertThat(entity.id).isNotZero
        assertThat(entity.message).isEqualTo("Some test message")
        assertThat(entity.createDate).isCloseTo(LocalDateTime.now(), Assertions.within(1, ChronoUnit.SECONDS))
        assertThat(entity.lastUpdateDate).isCloseTo(LocalDateTime.now(), Assertions.within(1, ChronoUnit.SECONDS))
    }
}